dataprofiler/profilers/utils.py

```diff
@@ -412,7 +413,19 @@ def __sub__(self: T, other: T) -> Any:
 T = TypeVar("T", bound=Subtractable)
 
 
+@overload
+def find_diff_of_numbers(
```

Comment:
Ended up going with an overload here. The issue is that there are a couple cases where we pass in a `Union[float, np.float64]` to `find_diff_of_numbers` and Unions don't satisfy the typevar bound.

Reply:
nice -- I would recommend formatting slightly different based on the docs I was seeing [here](https://docs.python.org/3/library/typing.html#typing.overload), but nice improvement

Reply:
actually, disregard this. I'm seeing it both ways

Reply:
Hmm, an object shouldn't be casted as `Union[float, np.float64]` but has the option to return as one or the other.

I wonder if this meant we did those returns wrong and should have done overloads on those funcs where it could return a float in one or np.float64 in the other.

Reply:
The main reason we have the `float` case is because `np.nan` is `float` and some of our methods could return `np.nan` if certain conditions aren't met. From what I can see, a lot of these conditions can't be solved by typing overloads (for example, some are based on class fields and not method parameters). 

At least for finding the diff between 2 `Union[float, np.float64]`, we shouldn't run into issues because `float` gets promoted to `float64` automatically.

Reply:
Seems to be this problem:
https://stackoverflow.com/questions/73011336/different-behavior-of-mypy-with-generic-class-in-python-3-8-and-3-10